hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/util/SparkKeyGenUtils.scala

```diff
@@ -65,10 +74,15 @@ object SparkKeyGenUtils {
     // For CustomKeyGenerator and CustomAvroKeyGenerator, the partition path filed format
     // is: "field_name: field_type", we extract the field_name from the partition path field.
     if (baseKeyGen.isInstanceOf[CustomKeyGenerator] || baseKeyGen.isInstanceOf[CustomAvroKeyGenerator]) {
-      typedProperties.getString(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME.key())
-        .split(",").map(pathField => {
-        pathField.split(CustomAvroKeyGenerator.SPLIT_REGEX)
-          .headOption.getOrElse(s"Illegal partition path field format: '$pathField' for ${baseKeyGen}")}).mkString(",")
+      val partitionFields = typedProperties.getString(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME.key())
+      if (includeType) {
+        partitionFields
+      } else {
+        partitionFields.split(",").map(pathField => {
```

Comment:
so the field type plain text is in format of the avro type name?

Reply:
field type is in the form of custom keygen types: simple or timestamp

Reply:
so the "field type" is not a notion of data type, let's make it more clear in the code.

Reply:
Addressed